R/features.R

```diff
@@ -0,0 +1,65 @@
+features_config <- new.env(parent = emptyenv())
+features_config[["known_features"]] <- tibble::tribble(
+  ~Name, ~RequiresPackage, ~Package,
+  "ndarray", TRUE, "ndarray",
+  "serde", TRUE, "serde",
+  "num-complex", TRUE, "num-complex",
+  "graphics", FALSE, NA
+)
+
+validate_extendr_features <- function(features, quiet) {
+  features <- features %||% character(0)
+  vctrs::vec_assert(features, character())
+  features <- unique(features)
+
+  unknown_features <- setdiff(features, features_config[["known_features"]][["Name"]])
+  unknown_features <- unknown_features[nzchar(unknown_features)]
```

Comment:
This is very much optional, but I just find it flows a tiny bit better
```suggestion
  unknown_features <- features %>% 
      setdiff(features_config$known_features$Name) %>%
      # Ignore empty features
      magrittr::extract(., nzchar(.))
```

Reply:
Fixed manually since github ui disallowed me to commit it from here. The `.` usage causes `{lintr}` to panic, and has no real value here, so I replaced last call by a custom `{vctrs}` - based method with a single arg. If pipelines succeed, I will merge this.

Reply:
Huh, that's weird, I don't see the style issue with that